From: Jeff Mahoney <jeffm@suse.com>
Subject: [PATCH 17/37] reiserfs: use better open options for internal files

 The generic read path will mark the file's atime during the course of the
 read. For xattr backing files, this behavior is undesirable, so we open
 the xattr files and directories with O_NOATIME. We also add the O_DIRECTORY
 flag where the xattr dir is opened for readdir.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>

--
 fs/reiserfs/xattr.c |    6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/fs/reiserfs/xattr.c	2007-05-10 07:58:27.000000000 -0400
+++ b/fs/reiserfs/xattr.c	2007-05-10 07:58:27.000000000 -0400
@@ -188,7 +188,7 @@ out:
 	dput(xadir);
 	if (err)
 		return ERR_PTR(err);
-	return dentry_open(xafile, NULL, O_RDWR);
+	return dentry_open(xafile, NULL, O_RDWR|O_NOATIME);
 }
 
 /*
@@ -686,7 +686,7 @@ int reiserfs_delete_xattrs(struct inode 
 		goto out;
 	}
 
-	fp = dentry_open(dir, NULL, O_RDWR);
+	fp = dentry_open(dir, NULL, O_RDWR|O_NOATIME|O_DIRECTORY);
 	if (IS_ERR(fp)) {
 		err = PTR_ERR(fp);
 		/* dentry_open dputs the dentry if it fails */
@@ -775,7 +775,7 @@ int reiserfs_chown_xattrs(struct inode *
 		goto out;
 	}
 
-	fp = dentry_open(dir, NULL, O_RDWR);
+	fp = dentry_open(dir, NULL, O_RDWR|O_NOATIME|O_DIRECTORY);
 	if (IS_ERR(fp)) {
 		err = PTR_ERR(fp);
 		/* dentry_open dputs the dentry if it fails */
